src/main/java/com/exactpro/th2/sailfish/utils/CompareFilter.java

```diff
@@ -0,0 +1,193 @@
+
+    }
+
+    @Override
+    public String getCondition() {
+        return stringFormatOperation + getValue();
+    }
+
+    @Override
+    public String getCondition(Object value) {
+        return value + getCondition() + getValue();
+    }
+
+    @Override
+    public Object getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean hasValue() {
+        return true;
+    }
+
+    private static Comparable<?> convertValue(String value) {
+        if (value.contains(String.valueOf(DecimalFormatSymbols.getInstance().getDecimalSeparator()))) {
+            return new BigDecimal(value);
+        }
+        return Long.parseLong(value);
+    }
+
+    private static Temporal convertDateValue(String value) {
+        if (value.contains(":")) {
+            if (value.contains("-")) {
+                return LocalDateTime.parse(value);
+            }
+            return LocalTime.parse(value);
+        }
+        return LocalDate.parse(value);
+    }
+
+    private boolean compareValues(Comparable<?> first, Comparable<?> second) {
+        int result;
+        if (first.getClass().equals(second.getClass())) {
+            if (first instanceof LocalDate) {
+                result = ((LocalDate)first).compareTo((LocalDate)second);
+            } else {
+                if (first instanceof LocalDateTime) {
+                    result = ((LocalDateTime)first).compareTo((LocalDateTime)second);
+                } else {
+                    if (first instanceof LocalTime) {
+                        result = ((LocalTime)first).compareTo((LocalTime)second);
+                    } else {
+                        if (first instanceof BigDecimal) {
+                            result = ((BigDecimal)first).compareTo((BigDecimal)second);
+                        } else {
+                            result = ((Long)first).compareTo((Long)second);
+                        }
+                    }
+                }
+            }
+        } else {
+            if (first instanceof LocalDate || first instanceof LocalDateTime || first instanceof LocalTime) {
+                throw new IllegalArgumentException(String.format("Failed to compare Temporal values {%s}, {%s}", first, second));
+            }
+            if (second instanceof LocalDate || second instanceof LocalDateTime || second instanceof LocalTime) {
+                throw new IllegalArgumentException(String.format("Failed to compare Temporal values {%s}, {%s}", first, second));
+            }
+            if (first instanceof BigDecimal) {
+                result = ((BigDecimal)first).compareTo(new BigDecimal(second.toString()));
+            } else {
+                if (second instanceof BigDecimal) {
+                    result = new BigDecimal(first.toString()).compareTo((BigDecimal)second);
+                } else {
+                    result = ((Long)first).compareTo((Long)second);
+                }
+            }
+        }
```

Comment:
I would suggest extracting the whole block into a method and use `return` in cases where we know what to do. It will allow you to simplify the conditions chain